src/mediaembed.js

```diff
@@ -37,6 +44,89 @@ export default class MediaEmbed extends Plugin {
 	static get pluginName() {
 		return 'MediaEmbed';
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		this._attachAutoEmbedingEvents();
+	}
+
+	/**
+	 * Attach events required for "auto-embeding" feature.
+	 *
+	 * @private
+	 */
+	_attachAutoEmbedingEvents() {
+		const editor = this.editor;
+		const modelDocument = editor.model.document;
+		const mediaRegistry = editor.plugins.get( MediaEmbedEditing ).registry;
+
+		let leftLivePosition, rightLivePosition;
+
+		// We need to listen on `Clipboard#inputTransformation` because we need to save positions of selection.
+		// After pasting a content, between those position can be located a URL that should be transformed to media.
+		this.listenTo( editor.plugins.get( Clipboard ), 'inputTransformation', () => {
+			const firstRange = modelDocument.selection.getFirstRange();
+
+			leftLivePosition = LivePosition.createFromPosition( firstRange.start );
+			leftLivePosition.stickiness = 'toPrevious';
+
+			rightLivePosition = LivePosition.createFromPosition( firstRange.end );
+			rightLivePosition.stickiness = 'toNext';
+		} );
+
+		modelDocument.on( 'change:data', () => {
+			if ( !leftLivePosition ) {
+				return;
+			}
+
+			const urlRange = new Range( leftLivePosition, rightLivePosition );
+			const walker = new TreeWalker( { boundaries: urlRange, ignoreElementEnd: true } );
+
+			let url = '';
+
+			for ( const node of walker ) {
+				if ( node.type === 'elementStart' ) {
+					return detach();
+				}
+
+				url += node.item.data;
+			}
+
+			// If the url does not match to universal url regexp, let's skip that.
+			if ( !url.match( URL_REGEXP ) ) {
```

Comment:
Paste the following "fusion" URL into the content: 
```
https://www.youtube.com/watch?v=H08tGjXNHO4https://open.spotify.com/album/2IXlgvecaDqOeF3viUZnPI?si=ogVw7KlcQAGZKK4Jz9QzvA
```

The following broken widget is created
![image](https://user-images.githubusercontent.com/1099479/45355338-69500400-b5c0-11e8-9a6f-7e80ed584099.png)

Something's wrong with discovering URLs in the pasted content.

Reply:
Also pasting `\s*https://www.youtube.com/watch?v=H08tGjXNHO4\s*` should result in auto-embed IMO. We mustn't assume users are white-space perfect when copy&pasting stuff from other places (e.g. other documents, some lists). That would be annoying from the UX point of view.

Reply:
When I paste `https://www.youtube.com/watch?v=H08tGjXNHO4https://open.spotify.com/album/2IXlgvecaDqOeF3viUZnPI?si=ogVw7KlcQAGZKK4Jz9QzvA` nothing happens.